python/ray/workflow/workflow_access.py

```diff
@@ -121,14 +123,40 @@ class LatestWorkflowOutput:
     step_id: "StepID"
 
 
+def _get_workflow_storage(workflow_id: str) -> workflow_storage.WorkflowStorage:
+    # Avoid initialize workflow in the workflow management actor,
+    # because it would invoke remote methods of itself,
+    # resulting in deadlock.
+    return workflow_storage.WorkflowStorage(workflow_id, _no_init_workflow=True)
+
+
+# TODO(suquark): what should be the default concurrency limit?
+@ray.remote(num_cpus=0)
+class WorkflowResultRespondingActor:
```

Comment:
I feel the API here is a little bit confusing, for example, it'll set the default value if it doesn't exist. If this happens, does this mean there is a bug that we haven't caught?


Is it possible to have only two API for simplicity:
   register_workflow -> ObjectRef
   set_output -> ObjectRef
?

Reply:
this comment is outdated